frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -167,23 +172,53 @@ function PlotlyFigure({
         // data is originalData + selectedpoints
         spec.data = data
         spec.layout.selections = selections
-
-        const hasSelectedPoints: boolean = spec.data.some(
-          (trace: any) =>
-            "selectedpoints" in trace && trace.selectedpoints.length > 0
-        )
-        if (hasSelectedPoints) {
-          spec.data.forEach((trace: any) => {
-            if (!trace.selectedpoints) {
-              // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
-              // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
-              trace.selectedpoints = []
-            }
-          })
+        if (spec.data) {
+          const hasSelectedPoints: boolean = spec.data.some(
+            (trace: any) =>
+              "selectedpoints" in trace &&
+              trace.selectedpoints &&
+              trace.selectedpoints.length > 0
+          )
+          if (hasSelectedPoints) {
+            spec.data.forEach((trace: any) => {
+              if (!trace.selectedpoints) {
+                // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
+                // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
+                trace.selectedpoints = []
+              }
+            })
+          }
         }
       }
     }
-
+    const rangeEvent = widgetMgr.getElementState(element.id, RANGE)
+    const autoRangeEvent = widgetMgr.getElementState(element.id, AUTORANGE)
+    const dragmode = widgetMgr.getElementState(element.id, DRAGMODE)
+
+    try {
+      if (rangeEvent && rangeEvent["xaxis.range[0]"]) {
```

Comment:
We only check for `xaxis.range[0]` here. Are the other keys definitely set if this is provided (e.g. `yaxis.range[0]`)?

Reply:
It seemed when I was doing testing it was always set and I can't imagine another way of doing so. From what I see [here](https://plotly.com/javascript/plotlyjs-events/#update-data), it seems like it's possible(?) that only the xrange checks and not the yrange so I'll add a separate check for the yrange.